src/MoBi.Presentation/UICommand/AddExpressionAsParameterStartValuesCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using MoBi.Core.Domain.Model;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Tasks.Interaction;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Presentation.UICommands;
+
+namespace MoBi.Presentation.UICommand
+{
+   public class AddExpressionAsParameterStartValuesCommand : ObjectUICommand<IMoBiProject>
+   {
+      private readonly IMoBiContext _context;
+      private readonly IInteractionTasksForExpressionProfileBuildingBlock _interactionTaskForExpressionProfileBuildingBlock;
+      private readonly IInteractionTasksForBuildingBlock<IParameterStartValuesBuildingBlock> _interactionTasksForPSVBuildingBlock;
+      private readonly IExpressionProfileToParameterStartValuesMapper _mapper;
+
+      public AddExpressionAsParameterStartValuesCommand(IMoBiContext context,
+         IInteractionTasksForExpressionProfileBuildingBlock interactionTaskForExpressionProfileBuildingBlock, IInteractionTasksForBuildingBlock<IParameterStartValuesBuildingBlock> interactionTasksForPSVBuildingBlock, IExpressionProfileToParameterStartValuesMapper mapper)
+      {
+         _context = context;
+         _interactionTaskForExpressionProfileBuildingBlock = interactionTaskForExpressionProfileBuildingBlock;
+         _interactionTasksForPSVBuildingBlock = interactionTasksForPSVBuildingBlock;
+         _mapper = mapper;
+      }
+
+      protected override void PerformExecute()
+      {
+         var expressionProfiles = _interactionTaskForExpressionProfileBuildingBlock.CreateFromPKML();
+
+         foreach (var expressionProfile in expressionProfiles)
```

Comment:
expressionProfiles.MapAllUsing(mapper).each(x=> _context.AddToHistory(_interactionTasksForPSVBuildingBlock.AddToProject(x));)

I am not sure how this is done for other types but this could be a MacroCommand (one action) instead of multples